sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -783,7 +783,7 @@ class SQLContext(@transient val sparkContext: SparkContext)
 
   /** Returns the specified table as a [[DataFrame]]. */
   def table(tableName: String): DataFrame =
-    DataFrame(this, catalog.lookupRelation(Seq(tableName)))
+    DataFrame(this, UnresolvedRelation(Seq(tableName)))
```

Comment:
My point here is we should leave the Relation Resolution for `SQLContext` or its extension,  through in most of case, the relation resolution resort to `catalog.lookupRelation`, but we never know if the logic will be changed or not, particularly for the extensions.  In #4784, it will returns an unresolved relation for `catalog.lookupRelation`, which required further analysis, it's maybe not the strong reason that we need to change code here, but it gives us an example that `catalog.lookupRelation` does not always act as the `eager analysis`.